### PR TITLE
Overwork

### PR DESCRIPTION
**Pull request by switchhax**
I saw that theses old PRs weren`t merge since a long time, because some complications arose. I tried to solve them.

 If you merge this PR, you can close #1690, #1795 and #1817

### CODE REVIEW

File: free-programming-books-zh.md [hunk 1]
```diff
@@ -29,6 +29,7 @@
   * [Android](#android)
   * [AWK](#awk)
   * [C/C++](#cc)
```
Comment (vhf) [inline free-programming-books-zh.md]: Could you please split this link like this:

```diff
-   * [C/C++](#cc)
+   * [C](#c)
+   * [C++](#c-1)
```

File: free-programming-books-zh.md [hunk 2]
```diff
@@ -29,6 +29,7 @@
...
+  * [C#](#C#)
```
Comment (vhf) [inline free-programming-books-zh.md]:
```diff
+  * [C#](#C#)
+  * [C#](#c-sharp)
```

File: free-programming-books-zh.md [hunk 3]
```diff
@@ -351,6 +352,11 @@
 * [跟我一起写Makefile(PDF)](http://scc.qibebt.cas.cn/docs/linux/base/%B8%FA%CE%D2%D2%BB%C6%F0%D0%B4Makefile-%B3%C2%F0%A9.pdf) (PDF)
 
 
+### C#
```
Comment (vhf) [inline free-programming-books-zh.md]:
```diff
-### C#
+### C Sharp
```

**Comment by vhf** [comment]
Many thanks! I added a few comments.

**Comment by switchhax** [comment]
Again tomorrow. Need to watch some EM stuff ...

**Comment by switchhax** [comment]
I make a seperate and new PR ;)

**Pull request closed without merging**